### PR TITLE
[MIG] account_invoice_production_lot: Migration to 10.0

### PR DESCRIPTION
**Pull request by cubells**
* Update README
* Update views
* Update translations
* Update demo data
* Update tests

cc @Tecnativa

### CODE REVIEW

File: account_invoice_production_lot/i18n/es.po [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+#. module: account_invoice_production_lot
+#: model:ir.model,name:account_invoice_production_lot.model_account_invoice_line
+msgid "Invoice Line"
+msgstr "Linea de factura"
+
+#. module: account_invoice_production_lot
+#: model:ir.model.fields,field_description:account_invoice_production_lot.field_account_invoice_line_order_line_ids
+msgid "Order Lines"
+msgstr "Lineas de factura"
```
Comment (pedrobaeza) [inline account_invoice_production_lot/i18n/es.po]: s/factura/pedido

File: account_invoice_production_lot/i18n/es.po [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+#. module: account_invoice_production_lot
+#: model:ir.model.fields,field_description:account_invoice_production_lot.field_account_invoice_line_prod_lot_ids
+#: model:ir.ui.view,arch_db:account_invoice_production_lot.view_invoice_line_form_prod_lots
+msgid "Production Lots"
+msgstr "Lotes de producción"
```
Comment (pedrobaeza) [inline account_invoice_production_lot/i18n/es.po]: Lotes/Nº de serie

File: account_invoice_production_lot/models/invoice_line.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
+# -*- coding: utf-8 -*-
```
Comment (pedrobaeza) [inline account_invoice_production_lot/models/invoice_line.py]: File should be named `account_invoice.py`

File: account_invoice_production_lot/models/invoice_line.py [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    _inherit = "account.invoice.line"
+
+    order_line_ids = fields.Many2many(
+        comodel_name='sale.order.line',
+        relation='sale_order_line_invoice_rel',
+        column1='invoice_id',
+        column2='order_line_id',
+        string='Order Lines',
+        readonly=True,
+        oldname='order_lines',
```
Comment (pedrobaeza) [inline account_invoice_production_lot/models/invoice_line.py]: As this is not a field with column, you don't need `oldname`. `relation`, `column1` and `column2` already point to the correct place to see for preserving data.

File: account_invoice_production_lot/models/invoice_line.py [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+    )
+
+    prod_lot_ids = fields.Many2many(
+        comodel_name='stock.production.lot',
+        relation='stock_prod_lot_invoice_rel',
```
Comment (pedrobaeza) [inline account_invoice_production_lot/models/invoice_line.py]: Being this a computed not stored field, you don't need to put `relation` nor `column1`

File: account_invoice_production_lot/__manifest__.py [hunk 6]
```diff
@@ -1,47 +1,28 @@
...
+    "version": "10.0.1.0.0",
+    "author": "Agile Business Group,Odoo Community Association (OCA)",
+    "summary": "Display delivered serial numbers in invoice",
+    'website': 'http://www.agilebg.com',
+    'license': 'AGPL-3',
+    'category': 'Accounting & Finance',
     "depends": [
         "account_accountant",
         "sale_stock",
         "stock_picking_invoice_link",
```
Comment (pedrobaeza) [inline account_invoice_production_lot/__manifest__.py]: There's no more need of `stock_picking_invoice_link` dependency, as you are fetching data from other place. Please remove this dependency.

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

File: account_invoice_production_lot/__manifest__.py (hunk 6 above)
Comment (cubells) [reply account_invoice_production_lot/__manifest__.py]: This line force me to add this dependency:

https://github.com/OCA/account-invoice-reporting/pull/61/files#diff-aa3ae9c1dd345dbc27d1368f7c18fd59R41

File: account_invoice_production_lot/models/account_invoice.py [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+    @api.multi
+    def _compute_prod_lots(self):
+        for line in self:
+            if not line.move_line_ids and not line.order_line_ids:
+                return
+            if line.move_line_ids:
+                line.prod_lot_ids = self.mapped(
+                    'move_line_ids.lot_ids')
+            else:
+                line.prod_lot_ids = self.mapped(
```
Comment (pedrobaeza) [inline account_invoice_production_lot/models/account_invoice.py]: Why don't you just simply use this and remove the dependency as stated. There's no other way now of creating invoices (you can't invoice from pickings), so this is enough.

File: account_invoice_production_lot/__manifest__.py (hunk 6 above)
Comment (cubells) [reply account_invoice_production_lot/__manifest__.py]: https://github.com/Tecnativa/account-invoice-reporting/blob/35f24d1b5f42f96425ef573f58dca84c140777d3/account_invoice_production_lot/models/account_invoice.py#L38

File: account_invoice_production_lot/models/account_invoice.py (hunk 7 above)
Comment (pedrobaeza) [reply account_invoice_production_lot/models/account_invoice.py]: Or is there any rough edge I don't find?

Comment (cubells) [reply account_invoice_production_lot/models/account_invoice.py]: Ok, I understand.

**Comment by pedrobaeza** [comment]
@luismontalba @chienandalu please review

**Review by pedrobaeza — APPROVED** [review]

File: account_invoice_production_lot/__manifest__.py [hunk 8]
```diff
@@ -1,47 +1,27 @@
...
-    'category': 'Generic Modules/Accounting',
-    "depends": [
-        "account_accountant",
-        "sale_stock",
-        "stock_picking_invoice_link",
-        ],
+    "version": "10.0.1.0.0",
     "author": "Agile Business Group,Odoo Community Association (OCA)",
     "summary": "Display delivered serial numbers in invoice",
     'website': 'http://www.agilebg.com',
```
Comment (luismontalba) [inline account_invoice_production_lot/__manifest__.py]: You could change website key following last convention:
https://github.com/OCA/maintainer-tools/blob/master/template/module/__openerp__.py#L9

**Review by luismontalba — APPROVED** [review]
Tested on runbot (minor comment in code)

**Review by chienandalu — APPROVED** [review]
Tested on runbot :+1:

**Pull request merged**